src/components/query-viewer-element/query-viewer-element.tsx

```diff
@@ -0,0 +1,108 @@
+import { Component, Prop } from "@stencil/core";
+
+@Component({
+  tag: "gx-query-viewer-element",
+  shadow: false
+})
+export class QueryViewerElement {
+  /**
+   * Name of the element
+   */
+  @Prop() name: string;
+  /**
+   * Title to show
+   */
+  @Prop() elementTitle: string;
+  /**
+   * How to show it
+   */
+  @Prop() visible: string;
+  /**
+   * Type of the element
+   */
+  @Prop() type: string;
+  /**
+   * Which axis, row or column
+   */
+  @Prop() axis: string;
+  /**
+   * Aggregation fucntion
+   */
+  @Prop() aggregation: string;
+  /**
+   * Data field
+   */
+  @Prop() dataField: string;
+  /**
+   * Type of the filter
+   */
+  @Prop() filterType: string;
+  /**
+   * Filter values comma separated
+   */
+  @Prop() filterValues: string;
+  /**
+   * Expand collapse type
+   */
+  @Prop() expandCollapseType: string;
```

Comment:
Looks that this is an enumerated list. We could take advantage of TypeScript to explicitly declare the list of accepted values. For example:

```
@Prop() expandCollapseType = "type1" | "type2" | "type3";
```